src/cmd-build

```diff
@@ -153,9 +158,13 @@ prepare_git_artifacts "${configdir_gitrepo}" "${PWD}/coreos-assembler-config.tar
 
 # These need to be absolute paths right now for rpm-ostree
 composejson=${PWD}/tmp/compose.json
+# Put this under tmprepo so it gets automatically chown'ed if needed
+lockfile_out=${tmprepo}/tmp/lockfile.json
```

Comment:
How about `manifest-lock.json`?

Reply:
It would be nice to have the output lockfile name indicate it is the output of the compose.. Something like `manifest-lockfile-out.json`. In practice we can then use `manifest-lockfile-in.json` as the input to the compose that we store in our fedora-coreos-configs git repo

Reply:
Not strongly against, though isn't where the lockfile located enough context already? E.g. if it's in the build dir, it's clearly an output from cosa, and if it's in the source config, it's clearly meant to be an input, right? And the commands will respect that by default (unless you explicitly pass e.g. `--update-lockfile` as suggested below).

Reply:
directory context does give you a lot, but if you copy things to different locations, discuss them in bug reports or on IRC, etc, you will lose that context. What if I want to copy two lockfiles to the same directory and then `vimdiff` them? All of these things can be done with them being the same filename and their directory setting the context, but I think It's way easier if we have explicit naming.

Reply:
Why would you copy them to the same directory?

Reply:
> Why would you copy them to the same directory?

`scp` both files from from remote system to local current working dir, download from web server into `~/Downloads/` folder etc.. Now I need to remember which one was input and which one was output. Not saying these are good reasons, but it happens and why not make it easier?

Are there any good reasons for making them be the exact same name?

Reply:
It sounds like you just want them named differently, but not necessarily explicitly labeled "input" and "output", which I don't particularly like since the format of the file is the exact same in either case. And e.g. a lockfile output can itself be used as an input.

How about `manifest-lock.json` for the one in the source config, and `ostree-lock.json` for the output one? That way the former is close to `manifest.yaml`, and the latter is close to the other `ostree-` objects.

Reply:
> It sounds like you just want them named differently

yes

> but not necessarily explicitly labeled "input" and "output"

i'd prefer it over arbitrary names. Of course the format is similar and one could use an output lockfile as input, but would be nice if the filename indicated where it came from or what it was used for. Looks like i'm not being very convincing. One final attempt :smile:  `lockfile.out.json` and `lockfile.in.json`

Reply:
another option.. the lockfile in the configs repo can just be `lockfile.json` but the generated output lockfile is `lockfile-out.json` or `lockfile-generated.json` or something that indicates it was an output artifact.

Reply:
I went with `manifest-lock.json` for input to follow the convention of using the same name as the manifest itself (could also match cargo/glide and do `manifest.lock` though the `.json` is helpful for e.g. web viewing), and `manifest-lock.generated.json` for the output so it's obvious that it's the same file format as the input one, but that it was generated from scratch.

Reply:
thanks - that works for me!